.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '44 17 * * 2'
```

Comment:
@prees1 Can you please help me understand why do we need this on a Cron job? and is it suppose to run every week (Tuesday)?

Reply:
Yea that doesn't seem needed. It is only there as the entire file was recommended from GitHub. It is otherwise configured to run on every push to either master or pull requests. So I can't see why it running periodically would identify anything novel.

Reply:
Removed it

Reply:
> Removed it

Scanning code on a schedule informs you about the latest vulnerabilities and errors that GitHub, security researchers, and the community discover, even when developers aren't actively maintaining the repository.
— [Configuring code scanning](https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#configuring-frequency)

Reply:
Good find! Thanks for checking @pshevtsov. So the rules that are being used to scan can actually be updated without anything touching the repo here. I will re-add it now for that benefit.